mtags/src/main/scala/scala/meta/internal/pc/Completions.scala

```diff
@@ -956,24 +956,29 @@ trait Completions { this: MetalsGlobal =>
         if (matchingType.size == 1) {
           s":${matchingType.head}"
         } else if (matchingType.size > 1) {
-          s"|${matchingType.mkString(",")}|"
+          s"|???,${matchingType.mkString(",")}|"
         } else {
           ":???"
         }
       }
 
       private def fillAllFields(): List[TextEditMember] = {
-        if (allParams.size > 1) {
+        val suffix = "autofill"
+        val shouldShow = allParams.exists(_.toString().startsWith(prefix))
+        val isExpilicitelyCalled = suffix.startsWith(prefix)
+
+        if ((shouldShow || isExpilicitelyCalled) && allParams.size > 1) {
           val editText = allParams.zipWithIndex
-            .map {
-              case (param, index) =>
+            .collect {
+              case (param, index) if !param.hasDefault =>
                 s"${param.name} = $${${index + 1}${findDefaultValue(param)}}"
             }
             .mkString(", ")
           val edit = new l.TextEdit(editRange, editText)
+          val filter = if (isExpilicitelyCalled) suffix else prefix + suffix
```

Comment:
When is filter not `"autofill"`?

Reply:
It's always `autofill`, at least part of it. This is because I wanted it to be the last option and not show up before any legit completions. The suffix in that case is not that important, it just needs to be something so that it goes later in the list.

Reply:
The sort text determines the order while the filter text determines what completion match what the user types. I'm struggling to understand the logic behind `if (isExpilicitelyCalled) suffix else prefix + suffix`, what is the difference between that and `"autofill"`?

Reply:
Ach right,  I was wrong, this was done to show autofill when parameters are completed.
Nothing actually to do with sorting, sorry!

Reply:
I would personally be in favor of not presenting autofill completions unless the user has typed a prefix of `"autofill"` or the empty string.

Reply:
Will leave it also when writing parameter names for now. I think it can be much more useful this way, otherwise nobody will find it. If that proves to be problematic, I will remove it myself.